tableauserverclient/models/property_decorators.py

```diff
@@ -84,3 +85,17 @@ def wrapper(self, value):
         return wrapper
 
     return property_type_decorator
+
+
+def property_matches(regex_to_match):
+
+    COMPILED_RE = re.compile(regex_to_match)
+
+    def wrapper(func):
+        @wraps(func)
+        def validate_regex_decorator(self, value):
+            if not COMPILED_RE.match(value):
+                raise ValueError("content_url can contain only ascii letters, numbers, dashes, and underscores")
```

Comment:
Make the message an argument that gets passed in because if we want to reuse this, then the error message would be wrong.